### PR TITLE
Add the ability to revoke instance keys

### PR DESCRIPTION
**Pull request by NobodyNada**
Adds a button to the key modal which regenerates the instance key.

### CODE REVIEW

File: app/views/bot_instances/_key_modal.html.erb [hunk 1]
```diff
@@ -8,6 +8,12 @@
         <code id="key-label">you should never see this</code>
       </div>
       <div class="modal-footer">
+        <%= form_tag "/you_should_never_see_this", id: "revoke-form", authenticity_token: false do %>
```
Comment (ArtOfCode-) [inline app/views/bot_instances/_key_modal.html.erb]: Any reason why you can't set the action from the view?

**Comment by NobodyNada** [comment]
There's only one modal, shared by each of the bot instances.  The "key" buttons contain data attributes which are written to the modal from JavaScript.

Since each instance has a different ID, I had to generate the action URL from JavaScript.  I guess a better solution would be providing the instance ID from a hidden field (or just not reusing the modal).

From Jonathan

**Comment by ArtOfCode-** [comment]
That sounds like something you can do with partial local variables. Let me dig a bit more.

**Comment by ArtOfCode-** [comment]
So it looks like you could remove the need to set the *bot* ID from JS, but not so easily the need to set the *instance* ID from JS. So yeah, not really worth doing one without the other.

**Pull request merged**